preparing for rc1.0.0

johannct opened the pull request:
This is a pull request to be merged when we are ready to cut v1.0.0

johannct committed:
first stab at PEGASE files reading

johannct committed:
fix tau logic

Commit:
Problem of label in zp versus zs

Commit:
Merge branch 'u/jct/rubicon2' of https://github.com/lephare-photoz/LEPHARE into u/jct/rubicon2

johannct committed:
Trigger CI

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/u/jct/rubicon2' into u/jct/rubicon2

johannct committed:
Merge pull request #380 from lephare-photoz/u/jct/rubicon2

new integration scheme, remove of resample and oneElLamda::ori. Major PR merging.

johannct committed:
precommit bug fix

johannct committed:
remove ori in oneElLambda constructor

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/381-we-need-to-implement-the-pegase-file-reading-code' into 381-we-need-to-implement-the-pegase-file-reading-code

johannct committed:
Merge pull request #388 from lephare-photoz/381-we-need-to-implement-the-pegase-file-reading-code

381 we need to implement the pegase file reading code.
No need to wait for this aside effort. Let's get rc1.0.0 rolling as the main branch to put some final effort before release

johannct committed:
remove read_flt with stream as input, and move
read_flt with filename as input from mag to flt.

johannct committed:
add a flt test file

johannct committed:
clean up a bit the LOG10/EXP10 constructs

johannct committed:
clean up and add tests for LOG10D/POW1OD

johannct committed:
generalize file convention for read_filters_doc
and add unit testing

johannct committed:
Merge pull request #390 from lephare-photoz/389-clean-read_flt-codebase

clean read_flt

johannct committed:
improve the tophat function. It is not used in
the codebase anymore, but remains useful in the
unit tests

johannct committed:
improve tophat filter. It is not used in the
code base anymore, but remains useful in the
unit tests

johannct committed:
small improvements to the doc

johannct committed:
work around doxygen/ruff collission

johannct committed:
remove macos-13

johannct committed:
generalize approximate_gaussian to larger window

johannct committed:
more tests

johannct committed:
attempts at bypassing coverage checking in some places

johannct committed:
remove unused bdincl and pypass some coverage checking

johannct committed:
fiddle with more coverage escapes and migrate pre-commit to avoid deprecation warning

johannct committed:
fiddle with more coverage escapes and migrate pre-commit to avoid deprecation warning

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/rc1.0.0' into rc1.0.0

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/rc1.0.0' into rc1.0.0

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/rc1.0.0' into rc1.0.0

johannct committed:
reorganize content

johannct committed:
use a script for coverage, so that it can be
also executed locally

johannct committed:
remove static library building from the
CMakeLists.txt

johannct committed:
remove static library building from the
CMakeLists.txt

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/393-cleaning-up-cmakeliststxt-and-fixing-up-an-issue-wrt-pypi' into 393-cleaning-up-cmakeliststxt-and-fixing-up-an-issue-wrt-pypi

johannct committed:
fix doc and comment to be consistent for the
units to be used in age file and range inputs

Commit:
modification to read the age selection in PEGASE

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/397-pegase2-sed-library-production-error2' into 397-pegase2-sed-library-production-error2

johannct committed:
change mention of PEGASE

johannct committed:
fix doc

johannct committed:
Merge pull request #394 from lephare-photoz/393-cleaning-up-cmakeliststxt-and-fixing-up-an-issue-wrt-pypi

393 cleaning up cmakeliststxt and fixing up an issue wrt pypi

Commit:
Implement some changes to use a structure of light SED

johannct committed:
change read_age name, turn it into a standalone
function, and specialize SEDLib<GalSED>
constructor, rather than using a condition

johannct committed:
add doc, improve comments, add some unit testing
for SEDLib code

johannct committed:
Merge pull request #398 from lephare-photoz/397-pegase2-sed-library-production-error2

397 pegase2 sed library production error2 : dealing with the ages for PEGASE2 is suboptimal but we will revisit it when a decision is made about incorporating more recent simulations.

johannct committed:
fix precommit

Commit:
Include also the light structure into generatePDF

Commit:
Gain a factor 2 in mag_gal with optimisation of to_tuple

Commit:
Optimize generate_PDF by limiting calls to log10

Commit:
Some cleaning

Commit:
correct the bad definition of the second color

Commit:
Changes suggested by Finlay

Commit:
Remove a cout

Commit:
remove a commented line

Commit:
expose lightLib in python

Commit:
Updated notebooks

johannct committed:
Merge branch 'main' into rc1.0.0

Commit:
Add two function to fit and analyse uncertainties for one source

Commit:
Add functionnalities to fit only one source without exposing the library

Commit:
Notebook with example of fitting one source

Commit:
Add the lsst filters for the tests

Commit:
Modify line length

Commit:
Add some printing

olivierilbert committed:
correcrt the test with the right sab

johannct committed:
doc improvements and cosmetics

johannct committed:
add coverage to readsource

Commit:
Define funz0 as a private variable of the class

Commit:
remove comments

olivierilbert committed:
Finally found the bottelneck which was creating a pause between each source. Still need to test in detail this version.

olivierilbert committed:
Forgot to pull the correction in funz0 before correcting the bottelneck

Commit:
Add Herschel filters for the tests

Commit:
Add Spitzer filters for tests

Commit:
Uncomment SEL_AGE from COSMOS.para to have access to the keyword

Commit:
Add tests incase of zfix, physical parameter computation with BC03

Commit:
Correct a bug for read keyword form run arguments

Commit:
Remove printed tests

Commit:
Correct problem in the notebook exemple

Commit:
Few minor modifications that I forgot to commit

Commit:
Add tests required on the filter range for colorRF

Commit:
Not working tests to retrieve the missing BC03 files

johannct committed:
fix the downloading of BC03 files for unit test and notebook

johannct committed:
fix the downloading of BC03 files for unit test and notebook

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/test_SED_to_SAO_FMarno' into test_SED_to_SAO_FMarno

johannct committed:
fix another unit test failure

johannct committed:
add printout to try to understand github test failure

Commit:
Change the selection age the ages

johannct committed:
exclude some coverage

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/test_SED_to_SAO_FMarno' into test_SED_to_SAO_FMarno

johannct committed:
exclude some coverage

olivierilbert committed:
Merge pull request #402 from lephare-photoz/test_SED_to_SAO_FMarno

Many changes:
- use a light library for the fitting function
- find a bottleneck when fitting each source
- add specific functions to fit an single source
- adapt the notebooks accordingly
- add usit tests

Commit:
Update the header of the library ascii file

Commit:
New tests on the FIR fit and rm_discrepant

Commit:
Improve a feature of stellar continuum substraction for the FIR fit

olivierilbert committed:
Minor modification in fit function argument for optimization

Commit:
Solve the problem when using restrict in fit. But not a great solution.

Commit:
Suggested change for lisibility

Commit:
use size_t rather rthan int for nlibext

Commit:
modify the size_t for nlibext

olivierilbert committed:
Merge pull request #410 from lephare-photoz/add_new_unit_tests

Add new tests and improvement in the FIR fit

Commit:
filter unit

Commit:
small adjustements

Commit:
add comment

olivierilbert committed:
Merge pull request #415 from lephare-photoz/filters_unit

filter unit

olivierilbert committed:
missing an initialisation for the model names in FIR

johannct committed:
pre commit

Commit:
Changes to allow the computation of the median LIR and associated uncertainties

Commit:
Add a test of LIR_MED

olivierilbert committed:
Merge pull request #417 from lephare-photoz/add_test_median_LIR

Changes to allow the computation of the median LIR and uncertainties

johannct committed:
remove nebv

johannct committed:
simplify some loops using size_t and removing
unnecessary variables

johannct committed:
purely cosmetic : remove unnecessary parentheses

johannct committed:
fix a bug in the for loops where references are
needed to avoid copying, which results in bug
when we loop to decrement values

johannct committed:
move the MW extinction where it is used

johannct committed:
remove nextlaw and simplify some expressions

johannct committed:
remove exposition of read_ext and def_zgrid, and
duplication of read_ext call

johannct committed:
turn FIR_SCALE into a long as it is a context

johannct committed:
cosmetics

johannct committed:
cosmetics

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/413-check-all-the-size_t---int-casts-and-remove-all-the-unnecessary-ones' into 413-check-all-the-size_t---int-casts-and-remove-all-the-unnecessary-ones

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/413-check-all-the-size_t---int-casts-and-remove-all-the-unnecessary-ones' into 413-check-all-the-size_t---int-casts-and-remove-all-the-unnecessary-ones

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/413-check-all-the-size_t---int-casts-and-remove-all-the-unnecessary-ones' into 413-check-all-the-size_t---int-casts-and-remove-all-the-unnecessary-ones

johannct committed:
remove unused variable

johannct committed:
remove unused variable

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/413-check-all-the-size_t---int-casts-and-remove-all-the-unnecessary-ones' into 413-check-all-the-size_t---int-casts-and-remove-all-the-unnecessary-ones

johannct committed:
remove the c++ historical executables, not updated anymore

johannct committed:
remove the c++ historical executables, not updated anymore

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/413-check-all-the-size_t---int-casts-and-remove-all-the-unnecessary-ones' into 413-check-all-the-size_t---int-casts-and-remove-all-the-unnecessary-ones

johannct committed:
remove declared but not instanciated method SED::fit_normalization

johannct committed:
remove imagm from fltUsed and fltUsedIR argument

johannct committed:
remove imagm as argument to fitIR

johannct committed:
remove imagm as argument to interp_lib

johannct committed:
avoid full SED copy in interp_lib

johannct committed:
add a function to create a static version of the
opacities vector.

johannct committed:
remove an unused call to read_opa

johannct committed:
extend the use of the static opa vector

Commit:
Change the age resolution for the selection

olivierilbert committed:
Merge pull request #420 from lephare-photoz/bug_closeage

Change the age resolution for the selection

Commit:
Change in the code for the emprical line ratios. Take the updated one by default.

Commit:
Add description of the empirical method for emission lines in a specialized notebook

Commit:
Update the tests accordingly

Commit:
Add the two missing files

johannct committed:
Merge pull request #418 from lephare-photoz/413-check-all-the-size_t---int-casts-and-remove-all-the-unnecessary-ones

413 check all the size t   int casts and remove all the unnecessary ones

Commit:
Add more documentation for emission lines

johannct committed:
Merge pull request #421 from lephare-photoz/emission_lines_empirical

Emission lines empirical

johannct committed:
fix precommit and add a helper function to
simplify plotting the spectrum of the fit result
for one source

johannct committed:
follow sentry and fix the thread unsafe situation for get_opa_vector

johannct committed:
add comment and plot curve

johannct committed:
make get_opa_vector python  with underscore

johannct committed:
try to exclude exception from coverage

johannct committed:
try to exclude exception from coverage

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/opa_cimplification' into opa_cimplification

johannct committed:
Merge pull request #422 from lephare-photoz/opa_cimplification

Opa simplification

johannct committed:
clean the beginning of globals

johannct committed:
Merge pull request #425 from lephare-photoz/423-use-constexpr-in-globalsh

clean the beginning of globals

Commit:
Add test on nz_prior

Commit:
binding nz_prior

Commit:
replace np.isclose by pytest.approx in the tests

Commit:
Tests on fitting several sources

olivierilbert committed:
Merge pull request #426 from lephare-photoz/add_more_tests

Add more tests

Commit:
Trying to fix mamba test error

Not sure why this started failing

Commit:
Trying further change to fix mamba cleanup

Commit:
Undoing changes which were causing new errors

raphaelshirley commented:
It seems like the new tests are much slower. Presumably because they use full SED lists and have larger z libraries. Is this strictly necessary? Will it not be sufficient to test on artificially small configs?

Commit:
Changed download logic to preference local files

If a file is present locally it will read target SEds from the file. If the user modifies the file locally those changes will therefore
propagate rather than overwrite it and redownload files

Commit:
Added exception handling to allow missing SED list files

Should we somehow allow all files to be missing inside a loop?

Commit:
Added more explicit warning test

Commit:
Removed now unnecessary extra files

this particular test will still be dependant on internet access but I guess that is fine since other do

johannct committed:
fix the corner case that was actually tested and was creating random fails from time to time

Commit:
Improve tests and allow writting the full chi2 for fit_onesource

johannct committed:
fix another corner case

johannct committed:
Merge pull request #429 from lephare-photoz/u/jct/bug_fix_sed_integrate

Improper treatment of corner case in SED::integrate

johannct committed:
Merge pull request #429 from lephare-photoz/u/jct/bug_fix_sed_integrate

Improper treatment of corner case in SED::integrate

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/rc1.0.0' into rc1.0.0

johannct committed:
fix another warning by sentry

johannct committed:
fix another warning by sentry

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/rc1.0.0' into rc1.0.0

Commit:
correct a bug in checking that .spec files exists

johannct committed:
fix a oops in a merge

Commit:
Add a print to have this test running on my laptop

Commit:
remove the problematic assert identified by sentry

olivierilbert committed:
Merge pull request #430 from lephare-photoz/new_test_readingsources

Improved tests and allows writting the full chi2 for fit_onesource

olivierilbert commented on scripts/run_coverage.sh:
maybe for @johannct

raphaelshirley replied:
Yes true. I guess people may want to run it locally if developing.

johannct replied:
I do that all the time....

Commit:
Small changes

Commit:
Make sentry happy

olivierilbert committed:
Merge pull request #436 from lephare-photoz/emlines_flux

    correct the bug detected by sentry on the outputed emission lines.
    add corresponding tests

johannct committed:
fix unwanted args

johannct committed:
remove argparse in profit of click. This should
solve the argument parsing problem

johannct committed:
remove argparse in profit of click. This should
solve the argument parsing problem

johannct committed:
Merge branch 'rc1.0.0' into 431-mag_abs-in-command-line

johannct committed:
Merge remote-tracking branch 'refs/remotes/origin/431-mag_abs-in-command-line' into 431-mag_abs-in-command-line

johannct committed:
add click to the requirements

johannct committed:
fix missing commits, and remove tests that use
sys.argv as this is not the way click works

johannct committed:
forgot to fix one config_file argument. This
cannot be set as None as it is supposed to be a
string. Default is now empty string

johannct committed:
./cov_html/index.html

Commit:
Change the MOD_EXTINC for stars, and avoid a warning

johannct committed:
sentry fix : typ argument's value must be uppercased

johannct committed:
fix a bug in opening config_file twice

johannct committed:
Merge pull request #437 from lephare-photoz/431-mag_abs-in-command-line

431 mag abs in command line

Commit:
Testing for local file overrides in place

If list files are changed remotely they will NOT override local files now! Is this definitely what we want?

Commit:
Further changes to path logic for local files

If the file is local read_file_list needs the full path not relative from the data repo

Commit:
Cleaned up test warnings and fixed repurcussions of logic change

Changing the file in one test meant that other tests broke as teh file had been changed so I delete it after the test

raphaelshirley committed:
Merge pull request #428 from lephare-photoz/issue/409/file-download-bug

Allow missing files

johannct committed:
Merge branch 'main' into rc1.0.0

Commit:
Small bug fix following new AUTO_ADAPT behaviour

compute offsets is always run and sets to zero if none are present and overwrites with sysshift if they are present

Tototime committed:
Add SCALE_STAR and QSO in c++ output

johannct committed:
improve corner case handling of sed::integrate

johannct committed:
this piece of code is actually useless

johannct committed:
update pre-commit config and run it

johannct committed:
reorder the new entries and fix a bug with SCALE_STAR and SCALE_QSO formatting

Commit:
check the size of a0. initiate at 0 if empty or uncorrect dimension

Commit:
Remove a comment

johannct committed:
Merge pull request #455 from lephare-photoz/433-make-sedintegrate-bullet-proof

improve corner case handling of sed::integrate

olivierilbert committed:
Merge pull request #456 from lephare-photoz/449-zphotarun_photoz-does-not-accept-empty-offset-list-anymore

check the size of a0. initiate at 0 if empty or uncorrect dimension

johannct committed:
fix dmmin initialization

johannct committed:
update alloutputkeys in the tests/data directory

johannct committed:
Merge branch 'rc1.0.0' into 448-add-possibility-to-retrieve-scale_star-scale_qso

johannct committed:
Merge pull request #450 from lephare-photoz/448-add-possibility-to-retrieve-scale_star-scale_qso

Add SCALE_STAR and QSO in c++ output

Commit:
Check that the value defined for omp_num_threads is meaningful

johannct committed:
add a stupid unit test

johannct committed:
fix the case where cpu_count() is not implemented

johannct committed:
fix the case where cpu_count() is not implemented

johannct committed:
Merge branch '366-notebooks-need-to-set-omp_num_threads' of https://github.com/lephare-photoz/lephare into 366-notebooks-need-to-set-omp_num_threads

olivierilbert committed:
Merge pull request #457 from lephare-photoz/366-notebooks-need-to-set-omp_num_threads

Check that the value defined for omp_num_threads is meaningful

Commit:
Add many comments in _spec.py, but no change in the structure of the code

olivierilbert committed:
Merge pull request #458 from lephare-photoz/306-document_specpy

comment the _spec.py file

johannct committed:
clean keyword

johannct committed:
Merge pull request #459 from lephare-photoz/u/jct/remove_split_code

clean keyword

johannct committed:
change name of onesource::substellar, add doc, and remove bindings

johannct committed:
Merge pull request #460 from lephare-photoz/252-understanding-onesourcesubstellar

change name of onesource::substellar, add doc, and remove bindings

johannct committed:
simplify time passing

johannct committed:
simplify time passing

johannct committed:
Merge branch 'remove_time_argument_in_zphota_write_outputs' of https://github.com/lephare-photoz/lephare into remove_time_argument_in_zphota_write_outputs

johannct committed:
Merge pull request #461 from lephare-photoz/remove_time_argument_in_zphota_write_outputs

simplify time passing

johannct committed:
changes not taken into account

johannct committed:
fix macos crash in calling time

johannct committed:
fix bug with time variable in photoz_lib

johannct committed:
fix bug with time variable in photoz_lib

johannct committed:
Merge branch 'rc1.0.0' of https://github.com/lephare-photoz/lephare into rc1.0.0

johannct committed:
fix a bug in runner management of verbose passed as arg

johannct committed:
remove boiler plate code and use functions instead, with the added benefit of a raise in case of unknown arg passed as arg to run()

johannct committed:
remove unnecessary args in sedtolib, that now raise

johannct committed:
Merge pull request #463 from lephare-photoz/462-bug-the-run-method-calls-upper-on-the-verbose-keyword-argument-without-checking-if-its-a-string-causing-a-crash-if-a-boolean-is-passed

remove boiler plate code and use functions instead

johannct committed:
bug fix : pass config_keymap to Runner init

johannct committed:
Merge pull request #465 from lephare-photoz/464-bug-the-zphota__init__-method-accepts-a-config_keymap-parameter-but-fails-to-pass-it-to-the-parent-class-constructor-causing-it-to-be-silently-ignored

bug fix : pass config_keymap to Runner init

Commit:
Bug fix allow process to accept MMEE inputs

raphaelshirley committed:
Merge pull request #467 from lephare-photoz/issue/466/meme_inputs_process

Bug fix allow process to accept MMEE inputs

johannct committed:
port the flag modifications initially developed
against main

johannct committed:
fix name change

johannct committed:
fix a corner case

johannct committed:
Merge pull request #475 from lephare-photoz/feature/pdf_flagger_rc

update to the flag computation

johannct committed:
Merge branch 'main' into rc1.0.0

johannct commented on src/lib/onesource.cpp:
```diff
@@ -2071,43 +2058,40 @@ void onesource::writeSpec(vector<SED *> &fulllib, vector<SED *> &fulllibIR,
   mIR = tmp.second;
   /*
     QSO CASE
   */
```
on the contrary the code now uses only angstrom everywhere

johannct commented on src/lephare/cli.py:
```diff
@@ -0,0 +1,44 @@
+        normalized = {k.upper(): v for k, v in kwargs.items() if v is not None}
+        runner = runner_class(
+            config_file=config,
+            **normalized,
```
fix in commit 6f14269

johannct commented on src/lib/globals.cpp:
```diff
@@ -104,41 +71,87 @@ double blackbody(double T, double lambda) {
+    ifstream stream;
+    // open the ascii file with all the opacity file listed
+    string opaListFile = lepharedir + "/opa/OPACITY.dat";
+    stream.open(opaListFile.c_str());
```
fix in commit 8feed96

johannct commented on src/lib/SED.cpp:
```diff
@@ -877,8 +630,7 @@ GalSED GalSED::generateEmSED(const string &emtype) {
+    auto nebular_contribution = add_neb_cont(qi[2]);  // Compute the continuum
     oneEm.generateEmPhys(zmet, qi[2]);  // Generate the emission lines
   } else if (emtype.compare("EMP_UV") == 0) {
     // Empirical method for emission lines
```
calc_ph is called in compute_luminosities, which is called in SEDLib.h.... Which means that the qi are precompted at the sedtolib stage. If a user wants to work their way out with a direct call to generateEmSED, they need to call calc_ph or compute_luminosity first. A condition check could be added as generateEmSED is called outside the very intensive make_maglib loop. Note that in order for the computation to be correct the template needs to be defined in the far UV, so only BC03 templates in effect allow for that currently.

johannct commented on src/lephare/runner.py:
```diff
@@ -34,195 +32,125 @@ class Runner:
+        if "typ" in self.config_keys:
+            if "typ" in kwargs:
+                self.typ = kwargs.pop("typ", self.typ)
+                self.typ = self.typ.upper()
```
No. The kwargs check is at L124 in Runner::run. It cannot be in _load_kwargs because "typ" is only relevant to a subset of execution classes, so its presence is just ignored if it is not in config_keys.
